### PR TITLE
Box border radius

### PR DESCRIPTION
**Pull request by spireken**

### CODE REVIEW

File: Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp [hunk 1]
```diff
@@ -30,22 +30,36 @@ using namespace Spire::Styles;
...
-      set_style(*box, std::move(style));
+      auto& border_size = get<int>("border-size", profile.get_properties());
+      auto& border_radius = get<int>("border-radius",
+        profile.get_properties());
+      auto update_style = [&, box = box] {
+        auto style = StyleSheet();
+        style.get(Any()).
+          set(BackgroundColor(QColor::fromRgb(255, 255, 255))).
+          set(border(border_size.get(), QColor::fromRgb(0xC8, 0xC8, 0xC8))).
+          set(Styles::border_radius(scale_width(border_radius.get()))).
```
Comment (shaobinspire) [inline Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp:48]: The Styles namespace is not needed here.

File: Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp [hunk 2]
```diff
@@ -30,22 +30,36 @@ using namespace Spire::Styles;
...
-        set(border_color(QColor::fromRgb(0xC8, 0xC8, 0xC8)));
-      set_style(*box, std::move(style));
+      auto& border_size = get<int>("border-size", profile.get_properties());
+      auto& border_radius = get<int>("border-radius",
+        profile.get_properties());
+      auto update_style = [&, box = box] {
+        auto style = StyleSheet();
+        style.get(Any()).
+          set(BackgroundColor(QColor::fromRgb(255, 255, 255))).
+          set(border(border_size.get(), QColor::fromRgb(0xC8, 0xC8, 0xC8))).
```
Comment (shaobinspire) [inline Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp:47]: The border width should be scaled.

File: Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp (hunk 1 above)
Comment (spireken) [reply Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp:48]: In this case it clashes with the UiProfile's border_radius property.

Comment (shaobinspire) [reply Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp:48]: Okay, I see.

**Review by shaobinspire — APPROVED** [review]

File: Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp [hunk 3]
```diff
@@ -30,22 +30,37 @@ using namespace Spire::Styles;
...
-        set(BackgroundColor(QColor::fromRgb(0xF5, 0xF5, 0xF5))).
-        set(border_color(QColor::fromRgb(0xC8, 0xC8, 0xC8)));
-      set_style(*box, std::move(style));
+      auto& border_size = get<int>("border-size", profile.get_properties());
+      auto& border_radius = get<int>("border-radius",
+        profile.get_properties());
+      auto update_style = [&, box = box] {
+        auto style = StyleSheet();
+        style.get(Any()).
+          set(BackgroundColor(QColor::fromRgb(255, 255, 255))).
```
Comment (kranar) [inline Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp:46]: This method of updating the stylesheet defeats the purpose of having a composable styling system. Only update the specific styling necessary to apply the change.

**Review by kranar — CHANGES_REQUESTED** [review]
Looks good.

**Pull request merged**